### PR TITLE
DEV: Plugin API method to replace post-menu buttons

### PR DESCRIPTION
**Pull request by markvanlan**
Reason described here: https://meta.discourse.org/t/post-voting-has-likes-enabled-but-they-wont-appear/248297/9?u=markvanlan

### CODE REVIEW

**Review by keegangeorge — APPROVED** [review]
This looks good, just a small note that the version number should be bumped and a changelog should be added.

See:
https://github.com/discourse/discourse/blob/5b6493ff4baf0cdb293aab53045545d8f9c01584/app/assets/javascripts/discourse/app/lib/plugin-api.js#L128-L132

File: app/assets/javascripts/discourse/app/lib/plugin-api.js [hunk 1]
```diff
@@ -656,6 +657,26 @@ class PluginApi {
     removeButton(name, callback);
   }
 
+  /**
+   * Replace existing button with a passed in widget
```
Comment (pmusaraj) [inline app/assets/javascripts/discourse/app/lib/plugin-api.js:661]:
```suggestion
   * Replace an existing button with a widget
```

**Review by pmusaraj — APPROVED** [review]
Let's give this a try. It's an edge case, but it makes sense in some cases to offer replacing a button directly (vs using `addPostMenuButton` and `removePostMenuButton`).

Make sure you add to changelong and bump the version, per @keegangeorge's comment, though.

File: app/assets/javascripts/discourse/app/lib/plugin-api.js [hunk 2]
```diff
@@ -129,7 +130,7 @@ import { _addBulkButton } from "discourse/components/modal/topic-bulk-actions";
 // based on Semantic Versioning 2.0.0. Please update the changelog at
 // docs/CHANGELOG-JAVASCRIPT-PLUGIN-API.md whenever you change the version
 // using the format described at https://keepachangelog.com/en/1.0.0/.
-export const PLUGIN_API_VERSION = "1.8.0";
+export const PLUGIN_API_VERSION = "1.8.1";
```
Comment (tgxworld) [inline app/assets/javascripts/discourse/app/lib/plugin-api.js:133]: @markvanlan Just a note that we're following Semantic versioning 2 and when you add functionality in a backwards compatible manner, you are supposed to bump the minor version and note the patch version. 

> Given a version number MAJOR.MINOR.PATCH, increment the:
> MAJOR version when you make incompatible API changes
> MINOR version when you add functionality in a backward compatible manner
> PATCH version when you make backward compatible bug fixes

**Pull request merged**